### ISSUE TITLE
False positive for `PIE796` files for custom enum_auto function

### ISSUE THREAD

**Issue by Skylion007** [astral-sh/ruff#8819]
The Nvidia cutlass library has it's own function `enum_auto` to populate enums as in `tools/library/scripts/pycutlass/src/pycutlass/library.py`
```
class StrideSupport(enum.Enum):
    Strided = enum_auto()
    Unity = enum_auto()
```
The linter currently complains with `PIE796` which it really should not unless it can be sure the output of enum_auto() is always the same: https://github.com/NVIDIA/cutlass/blob/b5d8a5d9ccb435268e2215144dca162b0fc6a826/python/cutlass_library/library.py#L49

**Comment by zanieb** [comment]
Hm... usually a repeated function call _would_ always yield the same results it's more of a special case that it wouldn't using global state. It seems like we'd have some false negatives if we were to allow any repeated function call here? I'm not sure I have a good solution otherwise though.

**Comment by Skylion007** [comment]
Can we check for global variable mutation inside of the function call somehow?

**Comment by Skylion007** [comment]
Yeah, at least enum.auto appears to work properly.

**Comment by charliermarsh** [comment]
Ahh I see, so `enum_auto` is a backwards-compatibility thing for Python 3.5. Interesting.

**Comment by zanieb** [comment]
Perhaps we could support this specific use-case by detecting cases where `enum.auto` is aliased as another name e.g. `enum_auto` then allow that name even if it's shadowed? Pretty weird though.

**Comment by zanieb** [comment]
We also have a weird `auto` method usage Prefect :D https://github.com/PrefectHQ/prefect/blob/d1894e8089d432cd62c7b63558fa75a8bf7049a7/src/prefect/utilities/collections.py#L41-L72

**Comment by charliermarsh** [comment]
I think this is gonna be hard to support as-is without a dedicated setting that requires users to provide an allow-list of functions or methods to treat as unique calls, but at the same time it feels a bit too niche for that. I'm going to close, but if folks have strong objections or other ideas we can reopen.
